executor/cook/executor.py

```diff
@@ -117,20 +119,24 @@ def launch_task(task):
             logging.warning('No command provided!')
             return None
 
-        process = subprocess.Popen(command, shell=True, stdout=sys.stdout.fileno(), stderr=sys.stderr.fileno())
+        process = subprocess.Popen(command,
+                                   bufsize=2*max_bytes_read_per_line,
```

Comment:
Could we be slowing down processes that block waiting for the executor to take data off the buffer?  Should we consider a larger buffer?

Reply:
Does `10* max_bytes_read_per_line` sound reasonable? The default is `io.DEFAULT_BUFFER_SIZE` which on my system is `8192`. `max_bytes_read_per_line` defaults to `4096`.

Reply:
I played around with `bufsize` locally.  When the subprocess is writing sufficiently large data (at least 64k, which is the default pipe buffer on my Linux system), the subprocess blocked no matter how big this buffer was, even if I set it to something huge like 1MB.  When the data is small (less than 64k), no blocking occurs, no matter the size of bufsize.

My conclusion is that `bufsize` should be set sufficiently large enough to transfer data from the pipe buffer to python's buffer without degrading performance.  We probably can avoid setting `bufsize` entirely, since `4096` seems like a reasonable default.

Reply:
What do you think about using a fixed size `Queue` and having one dedicated writer for `stdout` and `stderr`?

https://docs.python.org/3/library/queue.html

Reply:
We discussed offline and will not use the queue. I'll remove the default buffer size.